src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.util.RoleType;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+
+    public RoleDaoImpl(SessionFactory factory, Class<Role> clazz) {
+        super(factory, clazz);
+    }
+
+    @Override
+    public Optional<Role> getRoleByName(RoleType roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> getRoleByName = session.createQuery(
+                    "FROM Role  WHERE name = :name", Role.class);
+            getRoleByName.setParameter("name", roleName);
+            return getRoleByName.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Role with name " + roleName + " not found", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Error retrievign role with name " + roleName, e);
```
я б трохи змінив меседж щоб була явна різниця між цим і коли ти передаєш нейм якого не існує